### PR TITLE
Api 2.0

### PR DESCRIPTION
**Pull request by dns2utf8**
Fist steps for #30 and #40 

What do you think of the KeyboardButton::new() and From implementations? Is this the right way to go?

### CODE REVIEW

**Comment by dns2utf8** [comment]
Hi all

I am currently finshing a big paper. Can anyone take it over for now? I will have some more time in 5-6 weeks.

Regards.
@knsd @neuschaefer @rockneurotiko @LukasKalbertodt

File: src/types/mod.rs [hunk 1]
```diff
@@ -668,6 +679,56 @@ impl_encode!(ReplyKeyboardMarkup, 4,
...
+/// Telegram type "KeyboardButton" (directly mapped)
+#[derive(RustcDecodable, Debug, PartialEq, Clone)]
+pub struct KeyboardButton {
+    pub text: String,
+    pub request_contact: Option<bool>,
+    /// Note: request_contact and request_location options will only work in Telegram versions released after 9 April, 2016. Older clients will ignore them.
+    pub request_location: Option<bool>,
+}
+
+impl Default for KeyboardButton {
```
Comment (bjorn3) [inline src/types/mod.rs]: Maybe #[derive(Default)] for the struct.

File: src/types/mod.rs [hunk 2]
```diff
@@ -668,6 +679,56 @@ impl_encode!(ReplyKeyboardMarkup, 4,
...
+    fn default() -> Self {
+        KeyboardButton {
+            text: "".into(),
+            request_contact: None,
+            request_location: None,
+        }
+    }
+}
+//impl From<&'static str> for KeyboardButton {
```
Comment (bjorn3) [inline src/types/mod.rs]: Please remove this comment block.

File: src/types/mod.rs [hunk 3]
```diff
@@ -668,6 +679,56 @@ impl_encode!(ReplyKeyboardMarkup, 4,
     [0 => keyboard],
     [1 => resize_keyboard, 2 => one_time_keyboard, 3 => selective]);
 
+// ---------------------------------------------------------------------------
+/// Telegram type "KeyboardButton" (directly mapped)
+#[derive(RustcDecodable, Debug, PartialEq, Clone)]
+pub struct KeyboardButton {
+    pub text: String,
+    pub request_contact: Option<bool>,
```
Comment (bjorn3) [inline src/types/mod.rs]: Can this just be a bool?

**Comment by knsd** [comment]
Thanks for the pull request, but we are rewritting everything from scratch, so i'm closing it.

**Pull request closed without merging**